app-agent/src/utils.rs

```diff
@@ -0,0 +1,170 @@
+use std::collections::HashMap;
+
+// Damerau-Levenshtein distance
```

Comment:
Use triple slash to document function. Please add more details so that we understand what this is without googling.